overlord/state/taskrunner.go

```diff
@@ -142,14 +147,30 @@ func (r *TaskRunner) Ensure() {
 	}
 }
 
-// Stop stops all concurrent activities and returns after that's done.
-// Note that Stop will lock the state.
+// Stop kills all concurrent activities and returns after that's done.
 func (r *TaskRunner) Stop() {
-	r.state.Lock()
-	defer r.state.Unlock()
+	r.mu.Lock()
+	defer r.mu.Unlock()
 
-	for _, tb := range r.tombs {
+	for id, tb := range r.tombs {
 		tb.Kill(nil)
 		tb.Wait()
```

Comment:
not really about this branch but wondering if we should have two loops, one first Killin-ing all and then one waiting